spec/requests/post_controller_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:each) do
+      get '/users/:user_id/posts'
+    end
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'return 200' do
+      expect(response.status).to eq(200)
+    end
+    it 'renders the index template' do
+      expect(response).to render_template('index')
+    end
+    it 'include the list of users' do
+      expect(response.body).to include('All Posts')
+    end
+  end
+
+  describe 'GET /show' do
+    before(:each) do
+      get '/users/:user_id/posts/1'
+    end
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'return 200' do
+      expect(response.status).to eq(200)
+    end
+    it 'renders the show template' do
+      expect(response).to render_template('show')
+    end
+    it 'include the details of a post' do
+      expect(response.body).to include('Post')
+    end
+  end
+end
```

Comment:
- [ ] When I run the test file, most of them fail:

![image](https://github.com/lengushuru/Blog-app-on-rails/assets/77106962/519bd8ec-524d-4b48-8f35-bcb0a8fc8692)

They fail mostly because of route errors and testing the wrong response gotten from the page. I would advise you to properly declare the routes in the test file the same way you would in the browser and also test for the correct response gotten from each page. 👍 

#### Hint: For the response gotten from each page, alphabet casing matters. 😎 💻

Reply:
Thank you @Gambit142  for you review I will work on resolving the mentioned issues.